BtmsGateway/Middleware/MessageData.cs

```
@@ -156,6 +154,35 @@ public async Task PopulateResponse(HttpResponse response, RoutingResult routingR
         }
     }
 
+    private Dictionary<string, MessageAttributeValue> GetMessageAttributes(string? messageSubXPath)
+    {
+        var messageAttributes = new Dictionary<string, MessageAttributeValue>
+        {
+            {
+                Constants.MessageAttributeKeys.CorrelationId,
+                new MessageAttributeValue { StringValue = ContentMap.CorrelationId, DataType = "String" }
+            }
+        };
+
+        switch (messageSubXPath)
+        {
+            case Constants.SoapMessageTypes.ALVSClearanceRequest:
+                messageAttributes.Add(Constants.MessageAttributeKeys.MessageType,
```

Let's refactor to invoke the `messageAtrributes.Add` method once. Then the switch can just determine what the `StringValue` should be for the `MessageAttributeValue`. 

Let's either log and/or throw an exception if an unexpected SOAP message type is encountered

---

Done. This code is also executed for other message types so logging rather than throwing exception.